File: src/com/moneydance/modules/features/formula/split/Cell.java
package com.moneydance.modules.features.formula.split;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
public class Cell {
    String cell;
    char col;
    int row;;
    FormulaSplitTxn txn;
    List<String> deps;

    // check if the same dependency has been tried two or more times in a row
    public boolean isLoop() {
        int lastIndex = deps.size() - 1;
        return lastIndex > 1 && deps.get(lastIndex).equals(deps.get(lastIndex - 1));
    }

    public String getScript() {
        return txn.getCellSource(col);
    }
}

File: src/main/kotlin/com/moneydance/modules/features/formula/split/Cell.kt
package com.moneydance.modules.features.formula.split

data class Cell(
    val col: Char, // column starting from A
    val row: Int, // row number starting from 1
    val txn: FormulaSplitTxn,
    val deps: List<String> = ArrayList()
) {
    val cell: String
        get() {
            return "" + col + row
        }

    // check if the same dependency has been tried two or more times in a row
    val isLoop: Boolean
        get() {
            val lastTwo = deps.takeLast(2)
            return lastTwo.size == 2 && lastTwo.distinct().size == 1
        }
    val script: String?
        get() = txn.getCellSource(col)
}